File: Recursion_and_Backtracking_pepcoding_level_1/2.Recursion_on_array/1.display_arr.js
const display_arr = (arr, i = 0) => {
    if (i === arr.length) return;

    console.log(arr[i]);
    display_arr(arr, i + 1);
};

console.log(display_arr([-5, 34, 342, 34234, 32]));

// E
// 0 to  n

// F
// 1 to n

// F + E
// 0 + 1 to n

File: Recursion_and_Backtracking_pepcoding_level_1/2.Recursion_on_array/1.display_arr.rb
def display_arr arr, i=0
  return if i == arr.size

  puts arr[i]
  display_arr arr, i+1
end

display_arr [10,20,30,40,50]

# // E
# // 0 to  n

# // F
# // 1 to n

# // F + E
# // 0 + 1 to n
